### PR TITLE
Re-adds bagulo (remastered), 3 max bluespace cores, singulos may collapse if a third BoH is added

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- BoH + BoH = Singulo 
   - This time it has a unique animation associated with it spawning, it starts off as a tear and then becomes a singulo after a short time. 

- Max number of bluespace cores available to the crew reduced from 8 to 3

- Chucking a BoH into a singulo causes it to collapse in on itself. (Supermatter charged singulos are immune.)

## Why It's Good For The Game

BoH is generally harder to acquire now which means we can bring back soul

The max core number was reduced just to make sure it doesn't get spammed too much. Four's a crowd. 
It also might be the first time we actually hit a core limit

I thought it would be fun to make it so you can do something to stop it, so if you're a bad enough dude and can make a third BoH, you can chuck it into the bagulo to cause it to collapse. Singuloose generally ends the round regardless so I thought letting people have hero moments would be fun. 

## Changelog

:cl: Melbert
balance: Re-adds Bagulo
balance: The max number of bluespace cores available to the crew has been reduced to 3 (was: 8)
balance: Chucking a BoH into an uncharged singulo may save the station. 
/:cl:

### CODE REVIEW

**Comment by ATH1909** [comment]
is this immobile?

**Comment by EnterTheJake** [comment]
3 cores seems insanely harsh to me, considering bluespace is the anomaly with by far the most  amount of items you can craft, why even lower the amount when people barely make more 1 of these per shift?

**Comment by optimumtact** [comment]
> 3 cores seems insanely harsh to me, considering bluespace is the anomaly with by far the most amount of items you can craft.

The point of anomaly cores is to be hard limit and induce hard choices, but also you can move some of those items to other cores if you feel like you need more than 3.

**Comment by MrMelbert** [comment]
I still think it'd be fitting if we moved Phazon's BS core into the Ectoplasm core

**Comment by EnterTheJake** [comment]
> I still think it'd be fitting if we moved Phazon's BS core into the Ectoplasm core

Probably yeah, considering that phazon, doesn't really teleport, just phases through stuff.

**Comment by SkeletalElite** [comment]
Would be interesting if there was a way to scale anomaly cores with pop. 3 is restrictive but enough if there are only 30 people on who could potentially want to use those items. If there's 70 people on who could potentially want those items it could be restricting to the point of unfun for the people playing with those items in mind when they may only be able to actually make the item they're wanting once every 3 rounds.

**Comment by optimumtact** [comment]
I'm not going to complicate anomaly cores that much sorry, fixed limits they will remain.

**Comment by Sadboysuss** [comment]
this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!

**Comment by optimumtact** [comment]
> this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!

You need refined cores to use anything i'm pretty sure.

File: code/modules/power/singularity/singularity.dm [hunk 1]
```diff
@@ -320,6 +326,28 @@
...
+		message = span_danger("As [src] consumes [boh], it begins to collapse in on itself!"),
+		blind_message = span_hear("You hear aggressive crackling!"),
+		vision_distance = 15,
+	)
+	var/sound/collapse = sound('sound/effects/clockcult_gateway_disrupted.ogg')
+	collapse.pitch = -1
+	collapse.frequency = 0.5
+	playsound(loc, collapse, 200, vary = TRUE, extrarange = 3, falloff_exponent = 1, pressure_affected = FALSE, ignore_walls = TRUE, falloff_distance = 7)
+	addtimer(CALLBACK(src, PROC_REF(consume_boh_sfx)), 4 SECONDS)
+	animate(src, time = 4.5 SECONDS, transform = matrix(transform).Scale(0.25), flags = ANIMATION_PARALLEL, easing = ELASTIC_EASING)
```
Comment (LemonInTheDark) [inline code/modules/power/singularity/singularity.dm:343]: Transform is copy on modify so this is safe
```suggestion
	animate(src, time = 4.5 SECONDS, transform = transform.Scale(0.25), flags = ANIMATION_PARALLEL, easing = ELASTIC_EASING)
```

File: code/modules/power/singularity/singularity.dm [hunk 2]
```diff
@@ -320,6 +326,28 @@
...
+	animate(time = 0.5 SECONDS, alpha = 0)
+	QDEL_IN(src, 4.1 SECONDS)
```
Comment (LemonInTheDark) [inline code/modules/power/singularity/singularity.dm:345]: don't you mean 4.6

File: code/modules/power/singularity/singularity.dm [hunk 3]
```diff
@@ -320,6 +326,28 @@
...
+	name = "unstable [initial(name)]"
+	desc = "[initial(desc)] It seems to be collapsing in on itself."
+	visible_message(
+		message = span_danger("As [src] consumes [boh], it begins to collapse in on itself!"),
+		blind_message = span_hear("You hear aggressive crackling!"),
+		vision_distance = 15,
+	)
+	var/sound/collapse = sound('sound/effects/clockcult_gateway_disrupted.ogg')
+	collapse.pitch = -1
+	collapse.frequency = 0.5
```
Comment (LemonInTheDark) [inline code/modules/power/singularity/singularity.dm:340]: why set this on the sound and not in the frequency arg of playsound

File: code/modules/power/singularity/boh_tear.dm [hunk 4]
```diff
@@ -22,17 +16,29 @@
...
-		consume_range = BOH_TEAR_CONSUME_RANGE, \
-		grav_pull = BOH_TEAR_GRAV_PULL, \
+		consume_range = 1, \
+		grav_pull = 21, \
 		roaming = FALSE, \
 		singularity_size = STAGE_SIX, \
 	)
+	addtimer(CALLBACK(src, PROC_REF(bagulo_time)), 9 SECONDS, TIMER_DELETE_ME)
+	var/matrix/smaller = matrix(transform).Scale(0.25)
+	var/matrix/bigger = matrix(transform).Scale(2)
```
Comment (LemonInTheDark) [inline code/modules/power/singularity/boh_tear.dm:31]: similar on transforms

File: code/modules/power/singularity/boh_tear.dm [hunk 5]
```diff
@@ -22,17 +16,29 @@
...
 	AddComponent(
 		/datum/component/singularity, \
-		consume_range = BOH_TEAR_CONSUME_RANGE, \
-		grav_pull = BOH_TEAR_GRAV_PULL, \
+		consume_range = 1, \
+		grav_pull = 21, \
 		roaming = FALSE, \
 		singularity_size = STAGE_SIX, \
 	)
+	addtimer(CALLBACK(src, PROC_REF(bagulo_time)), 9 SECONDS, TIMER_DELETE_ME)
```
Comment (LemonInTheDark) [inline code/modules/power/singularity/boh_tear.dm:29]: shouldn't this be 10 seconds?

Comment (LemonInTheDark) [reply code/modules/power/singularity/boh_tear.dm:29]: or do you want it to delete like as it fades kinda thing (so like 9.5)

File: code/modules/power/singularity/singularity.dm (hunk 2 above)
Comment (LemonInTheDark) [reply code/modules/power/singularity/singularity.dm:345]: or like idk 4.3

**Review by LemonInTheDark — CHANGES_REQUESTED** [review]
I am so on board with this you have no idea

**Comment by Astrogem2** [comment]
so you can't refine more than 3 bluespace cores per round is what i'm picking up, seems fair i guess considering the new additions if added.

**Comment by TagGamerGang22** [comment]
Something important to note is that cores can be ordered for free roundstart by scientists via their departmental order. If they're the RD then they get a ~6k departmental budget to spend roundstart which they can immediately spend on a couple BP cores.

**Comment by YuliceTheDruid** [comment]
If we're hard limiting cores that much can we perhaps not require atmos knowledge to refine them

**Comment by Sadboysuss** [comment]
> If we're hard limiting cores that much can we perhaps not require atmos knowledge to refine them

no

**Comment by Sadboysuss** [comment]
> > this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!
> 
> You need refined cores to make anything i'm pretty sure.
> 
> The design document is clear that it's a hard limit that must be obeyed by everything, so if there are ways to go over the cap, that's a bug.



> > this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!
> 
> You need refined cores to make anything i'm pretty sure.
> 
> The design document is clear that it's a hard limit that must be obeyed by everything, so if there are ways to go over the cap, that's a bug.

fixing an anomaly drops a refined core

**Comment by LemonInTheDark** [comment]
> fixing an anomaly drops a refined core

I have been saying this for years (except when I am told I am wrong) and did anyone listen?!!?

File: code/modules/power/singularity/singularity.dm (hunk 2 above)
Comment (MrMelbert) [reply code/modules/power/singularity/singularity.dm:345]: Some some reason probably related to animates being client side making the `qdel` timer equal to or longer than the animate timers results in a noticeably long wait between the animation finishing and the qdel happening. 

So I just tweaked the timers down until they looked good (which is where they are now)

File: code/modules/power/singularity/boh_tear.dm (hunk 5 above)
Comment (MrMelbert) [reply code/modules/power/singularity/boh_tear.dm:29]: See above

**Comment by optimumtact** [comment]
> > > this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!
> > 
> > 
> > You need refined cores to make anything i'm pretty sure.
> > The design document is clear that it's a hard limit that must be obeyed by everything, so if there are ways to go over the cap, that's a bug.
> 
> > > this restriction is only on fabricating them via ordnance, spawned anomaly cores can still go over 3, better get that anomaly!
> > 
> > 
> > You need refined cores to make anything i'm pretty sure.
> > The design document is clear that it's a hard limit that must be obeyed by everything, so if there are ways to go over the cap, that's a bug.
> 
> fixing an anomaly drops a refined core

4e6

**Comment by Holinka4ever** [comment]
Remember, fauna in ice box have a drop chance of bluespace raw core,  it will be really yeaks if you wanted to make bagulo, but you cant due of bs core being lost in ice mine.

**Pull request merged**